### PR TITLE
feat: move reth genesis to alloy-genesis

### PR DESCRIPTION
**Pull request by supernovahs**
closes https://github.com/paradigmxyz/reth/issues/6085

### CODE REVIEW

File: crates/genesis/src/account.rs
Comment (mattsse) [inline crates/genesis/src/account.rs:1]: I don't think we need this type,

this is the same as 

https://github.com/paradigmxyz/reth/blob/7c9949962b97a6e3cb230e82a44c91ce8d0b1c98/crates/primitives/src/account.rs#L14-L21

right?

File: crates/genesis/src/genesis.rs
Comment (mattsse) [inline crates/genesis/src/genesis.rs:1]: this should be moved to lib.rs

**Review by mattsse — CHANGES_REQUESTED** [review]
great!

a few more nits, then this is good

File: crates/genesis/src/genesis.rs [hunk 1]
```diff
@@ -0,0 +1,1479 @@
...
+impl From<GenesisAccount> for Account {
+    fn from(value: GenesisAccount) -> Self {
+        Account {
+            // nonce must exist, so we default to zero when converting a genesis account
+            nonce: value.nonce.unwrap_or_default(),
+            balance: value.balance,
+            bytecode_hash: value.code.map(keccak256),
+        }
+    }
+}
```
Comment (supernovahs) [inline crates/genesis/src/genesis.rs:189]: @mattsse Deleting Account type would mean , we would have to import Account from reth -primitives? That's not an option i think right?

Comment (mattsse) [reply crates/genesis/src/genesis.rs:189]: we don't need this conversion anymore, instead we can move this to reth-primitives when we replace the genesis type there

**Review by mattsse — APPROVED** [review]
nice, as next step we can also replace the genesis types in 

https://github.com/alloy-rs/alloy/blob/main/crates/node-bindings/src/genesis.rs

with this one

**Pull request merged**